domain/schema/secret.go

```diff
@@ -183,19 +183,26 @@ CREATE TABLE secret_revision (
     uuid TEXT PRIMARY KEY,
     secret_id TEXT NOT NULL,
     revision INT NOT NULL,
+    CONSTRAINT fk_secret_revision_secret_metadata_id
+        FOREIGN KEY (secret_id)
+        REFERENCES secret_metadata (secret_id)
+);
+
+CREATE UNIQUE INDEX idx_secret_revision_secret_id_revision ON secret_revision (secret_id,revision);
+
+CREATE TABLE secret_revision_obsolete (
+    revision_uuid TEXT PRIMARY KEY,
     obsolete BOOLEAN NOT NULL DEFAULT (FALSE),
     -- pending_delete is true if the revision is to be deleted.
     -- It will not be drained to a new active backend.
     pending_delete BOOLEAN NOT NULL DEFAULT (FALSE),
     create_time DATETIME NOT NULL DEFAULT (STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW', 'utc')),
     update_time DATETIME NOT NULL DEFAULT (STRFTIME('%Y-%m-%d %H:%M:%f', 'NOW', 'utc')),
```

Comment:
We don't want create time here, and we would need to ensure update time is used where needed when filling out revision metadata struct. This is assuming we lazily insert a row in this table only when needed.